### PR TITLE
Add black box option to context menu for source tree items and source header

### LINKED ISSUE #7287: Context menu Source actions should be available from the SourcePanel and Source Tab
Someone was asking on IRC how one could blackbox a script, so I tried myself by: 
- doing a right click on the sources panel, on the script I wanted to blackbox: -> no luck
- opening the source, right click on the source tab -> no luck
- right click *in* the source content -> jackpot

So this is pretty confusing to me that we have to open a source and specifically right-click in the content.
We should be able to perform any source-related action from whatever represent a source in the UI (so, from the sources panel and the source header, maybe from the callstack as well)

I think it's pretty important because at the moment users might miss that we have such feature

### PR DESCRIPTION
**Pull request by derek-li**
Fixes #7287 

Summary of changes:
- Adds black box option to source header and source items in the tree

Demo:
Source header GIF: https://gyazo.com/441defadd22762e5c3142011241b8e8b
Source tree GIF: https://gyazo.com/5c8354f69db18a3ffd226448d8334006

### CODE REVIEW

File: src/components/Editor/Tab.js [hunk 1]
```diff
@@ -135,13 +137,24 @@ class Tab extends PureComponent<Props> {
       }
     ];
 
-    items.push({
-      item: {
-        ...tabMenuItems.prettyPrint,
-        click: () => togglePrettyPrint(tab),
-        disabled: !shouldShowPrettyPrint(source)
+    items.push(
```
Comment (darkwing) [inline src/components/Editor/Tab.js]: I think we want to add `shouldBlackbox` here since we use it for other places.

**Comment by darkwing** [comment]
Functionally it works well, just implement `shouldBlackBox` and I think we will be set!

File: src/components/Editor/Tab.js (hunk 1 above)
Comment (derek-li) [reply src/components/Editor/Tab.js]: Added the changes!

**Comment by derek-li** [comment]
Appreciate the review @darkwing :)

**Comment by darkwing** [comment]
@derek-li Looks like you need to do a `yarn jest -u` to update snapshots, then I think we're good!

**Comment by derek-li** [comment]
Hey @darkwing, ran the jest updates, but it looks like more CircleCI tests are failing than before. Do you have any input?

**Comment by derek-li** [comment]
Not sure what changed, but the black box option on the sourcestreeitem context menu stopped working. Currently investigating.

**Comment by darkwing** [comment]
How's the investigation going?  I noticed that the blackbox source context menu item is always disabled, so we need a bit more work here.

**Comment by derek-li** [comment]
Thanks for checking in. Sorry for being a bit slow, still a noob :)

I found that the sources prop in the sourcesTree is passing in state.sources.relativeSources. The problem with that is relativeSources doesn't seem up to date with state.sources.sources.

So even when a source in state.sources.sources has attributes `loadedState: "loaded"` and `isBlackBoxed: true`, we get this as the sources prop:
0d13964fd413ec4da49c1b0602d9d12e

which results in the button being disabled, as well as the blackbox icon not being rendered in the sources tree panel.

I'm currently thinking of potential solutions and am strongly considering using another selector. Do you know why we are using relativeSources in this case?

**Comment by derek-li** [comment]
@darkwing I think this should be good to merge. 
There's just the issue with the inexact source type. Do you know what's causing the error?

**Comment by jasonLaster** [comment]
looks like the flow issues are unrelated. I believe rebasing should fix them

**Comment by derek-li** [comment]
Okay looking into it now.

**Comment by darkwing** [comment]
@derek-li The source flow issue can be fixed with:

```diff
$ git diff
diff --git a/src/components/PrimaryPanes/SourcesTreeItem.js b/src/components/PrimaryPanes/SourcesTreeItem.js
index ac5e58017..2f64d2d9f 100644
--- a/src/components/PrimaryPanes/SourcesTreeItem.js
+++ b/src/components/PrimaryPanes/SourcesTreeItem.js
@@ -246,7 +246,7 @@ class SourceTreeItem extends Component<Props, State> {
         className={classnames("node", { focused })}
         key={item.path}
         onClick={this.onClick}
-        onContextMenu={e => this.onContextMenu(e, item, source)}
+        onContextMenu={e => source && this.onContextMenu(e, item, source)}
       >
         {this.renderItemArrow()}
         {this.getIcon(item, depth)}
```

**Comment by derek-li** [comment]
Thanks for the help guys!

Re based and added the tweak

**Pull request merged**